src/core.c/CompUnit/Repository/Installation.rakumod

```diff
@@ -119,34 +120,34 @@ class CompUnit::Repository::Installation does CompUnit::Repository::Locally does
         self!prefix-writeable || (!$.prefix.e && ?$.prefix.mkdir)
     }
 
-    method !config(:$wrapper-mode) {
+    method !seed-config(:$wrapper-mode) {
+        %!config-seed =
+            wrapper-mode => $wrapper-mode // 'path',
+        ;
+    }
+
+    method !write-config() {
+        self!config-file.spurt(Rakudo::Internals::JSON.to-json(%!config));
+    }
+
+    method !config() {
         unless %!config {
             my $file = self!config-file;
             if $file.e {
                 %!config = Rakudo::Internals::JSON.from-json($file.slurp);
             }
-            elsif self!prefix-writeable {
-                $file.spurt(Rakudo::Internals::JSON.to-json(
-                    %(
-                        wrapper-mode => $wrapper-mode // 'path',
-                    )
-                ));
-                %!config = Rakudo::Internals::JSON.from-json($file.slurp);
-            }
             else {
-                %!config =
-                    wrapper-mode => $wrapper-mode // 'path',
-                ;
+                %!config = %!config-seed;
+                self!write-config if self!prefix-writeable;
             }
         }
         %!config
     }
 
     method change-wrapper-mode($mode) {
```

Comment:
I think we'd want to mark any publicly exposed methods as an implementation detail

Reply:
Agreed! I still have to learn many things about core hacking. :-)

Reply:
Done.

Reply:
Done.